protocol-gateway/mqtt-protocol-gateway-template/src/main/java/org/eclipse/hono/gateway/sdk/mqtt2amqp/MultiTenantConnectionManager.java

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.gateway.sdk.mqtt2amqp;
+
+import java.util.function.Consumer;
+
+import org.apache.qpid.proton.message.Message;
+import org.eclipse.hono.client.MessageConsumer;
+import org.eclipse.hono.client.device.amqp.AmqpAdapterClientFactory;
+import org.eclipse.hono.client.device.amqp.CommandResponder;
+import org.eclipse.hono.client.device.amqp.EventSender;
+import org.eclipse.hono.client.device.amqp.TelemetrySender;
+import org.eclipse.hono.config.ClientConfigProperties;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.mqtt.MqttEndpoint;
+
+/**
+ * Manages connections for multiple tenants.
+ */
+public interface MultiTenantConnectionManager {
+
+    /**
+     * Connect to Hono's AMQP adapter with the given configuration.
+     *
+     * @param tenantId The tenant to connect.
+     * @param vertx The Vert.x instance to use for the connection.
+     * @param clientConfig The configuration of the connection.
+     * @return a succeeded future if the connection could be established within the time frame configured with
+     *         {@link ClientConfigProperties#getConnectTimeout()}, a failed future otherwise.
+     */
+    Future<Void> connect(String tenantId, Vertx vertx, ClientConfigProperties clientConfig);
+
+    /**
+     * Adds an MQTT endpoint for the given tenant.
```

Comment:
I think it would be good to mention that `connect` has to have been called here first for the tenant.